src/Datadog.Trace.ClrProfiler.Managed.Loader/Startup.cs

```diff
@@ -14,20 +15,48 @@ public class Startup
         /// </summary>
         static Startup()
         {
+            AppDomain.CurrentDomain.AssemblyResolve += ResolveManagedProfiler;
             LoadManagedAssembly();
         }
 
         internal static bool ManagedAssemblyFound { get; set; }
 
+        private static Assembly ResolveManagedProfiler(object sender, ResolveEventArgs args)
+        {
+            string tracerDirectory = Environment.GetEnvironmentVariable("DD_DOTNET_TRACER_HOME");
+
+            if (!string.IsNullOrEmpty(tracerDirectory))
+            {
+#if NETSTANDARD2_0
+                string tracerFrameworkDirectory = "netstandard2.0";
+#else
+                string tracerFrameworkDirectory = "net45";
```

Comment:
We may want a directory for 461+ because of the Sigil-vNext conditional references. Thoughts?

Reply:
.NET Framework 4.6.1 implements .NET Standard 2.0, so we can load the version targeting `netstandard2.0` in an application running on `net461` or above. The version targeting `net45` is only for apps running on .NET Framework 4.5 - 4.6.

The only reason we target `net461` separately in our NuGet packages is because [it's recommended](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/cross-platform-targeting#multi-targeting) to support devs using older tooling:

> CONSIDER adding a target for `net461` when you're offering a `netstandard2.0` target.
>
> Using .NET Standard 2.0 from .NET Framework has some issues that were addressed in .NET Framework 4.7.2. You can improve the experience for developers that are still on .NET Framework 4.6.1 - 4.7.1 by offering them a binary that is built for .NET Framework 4.6.1.

Reply:
We still need to add the run-time checks to load the correct assembly, though.

Reply:
After Lucas's comments, is the sentiment that we should detect the .NET Framework version and use `netstandard2.0` if we're on .NET Framework 4.6.1+? If so, do you guys have a preference on which of these mechanisms to use to identify the framework version? Just checking the CLR version doesn't give us enough info, which is why all these other solutions exist: https://stackoverflow.com/questions/951856/is-there-an-easy-way-to-check-the-net-framework-version

Reply:
Sorry, after some research, I realized that what I suggested is probably a bad idea. Even though `net461` supports `netstandard2.0`, it does so by deploying additional libraries with the app, because `net461` does _not_ implement all of the APIs in `netstandard2.0`. See [this article](https://devblogs.microsoft.com/dotnet/net-framework-4-7-1-runtime-and-compiler-features/ ) for more details.

Reply:
Ok, I've updated the logic here to use either the `net45` assets or the `net461` assets for .NET Framework apps. Let me know what your thoughts are on the implementation.